libtransmission/torrent.cc

```diff
@@ -817,6 +817,53 @@ static void torrentInit(tr_torrent* tor, tr_ctor const* ctor)
     }
 }
 
+static void tr_torrentAddDefaultTrackers(tr_torrent* tor)
+{
+    std::list<std::string> trackerURLs = {};
+    int numExistingTrackers = tr_torrentTrackerCount(tor);
+    int numNewTrackers = tor->session->defaultTrackersList.size();
+
+    if (!numNewTrackers || tor->isPrivate())
+    {
+        return;
+    }
+
+    // copy existing tracker URLs
+    for (int i = 0; i < numExistingTrackers; ++i)
+    {
+        auto tracker = tr_torrentTracker(tor, i);
+        trackerURLs.push_back(tracker.announce);
+    }
+
+    // add the new ones
+    for (std::string_view url : tor->session->defaultTrackersList)
+    {
+        if (tr_urlIsValidTracker(url))
+        {
+            // check for duplicates
+            bool duplicate = false;
+            for (auto trackerURL : trackerURLs)
+            {
+                if (trackerURL == url)
+                {
+                    duplicate = true;
+                    break;
+                }
+            }
+
+            if (duplicate)
+            {
+                continue;
+            }
+
+            tor->announceList().add(url);
+        }
+    }
```

Comment:
`tr_announce_list.add()` already ignores duplicates, so no need to check them here

```suggestion
    if (tor->isPrivate())
    {
        return;
    }
    
    for (auto const& url : tor->session->defaultTrackersList)
    {
            tor->announceList().add(url);
    }
```